pkg/slices/slices.go

```diff
@@ -158,6 +158,9 @@ func Map[E any, O any](s []E, f func(E) O) []O {
 // MapFilter runs f() over all elements in s and returns any non-nil results
 func MapFilter[E any, O any](s []E, f func(E) *O) []O {
 	n := make([]O, 0, len(s))
+	if f == nil {
+		return n
+	}
```

Comment:
Can we revert these? I don't think we should make `nil` a valid input

Reply:
Sure, happy to. I added it because a `nil` input in `Flatten` causes a nil-pointer panic, and I thought it would be good to handle that, which then led me to adding it to `MapFilter` for consistency. But if that's not what you'd prefer, I'll leave it as is.

Reply:
for Flatten I think its reasonable but not for MapFilter

Reply:
Ok then, I'll leave Flatten in and remove the nil handler for MapFilter.